### ISSUE TITLE
Unable to start activity MoPubFullscreenActivity android.view.InflateException

### ISSUE THREAD

**Issue by m-melis** [mopub/mopub-android-sdk#466]
Just found the following error in Crashlytics: `Unable to start activity ComponentInfo{com.tmsp.mappacolorizone/com.mopub.mobileads.MoPubFullscreenActivity}: android.view.InflateException: Binary XML file line #24: Binary XML file line #24: Error inflating class android.widget.ImageView`

On a Huawei Nexus 6P Android 8.1.0.

#### Evidence:
```
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.xxx.yyy/com.mopub.mobileads.MoPubFullscreenActivity}: android.view.InflateException: Binary XML file line #24: Binary XML file line #24: Error inflating class android.widget.ImageView
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2778)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
       at android.app.ActivityThread.-wrap11(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6494)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```
```
Caused by android.content.res.Resources$NotFoundException: Unable to find resource ID #0x7f0800ce
       at android.content.res.ResourcesImpl.getResourceName(ResourcesImpl.java:253)
       at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:760)
       at android.content.res.ResourcesImpl.loadDrawable(ResourcesImpl.java:630)
       at android.content.res.Resources.loadDrawable(Resources.java:886)
       at android.content.res.TypedArray.getDrawableForDensity(TypedArray.java:953)
       at android.content.res.TypedArray.getDrawable(TypedArray.java:928)
       at android.widget.ImageView.<init>(ImageView.java:188)
       at android.widget.ImageView.<init>(ImageView.java:171)
       at android.widget.ImageView.<init>(ImageView.java:167)
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:334)
       at android.view.LayoutInflater.createView(LayoutInflater.java:647)
       at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:58)
       at android.view.LayoutInflater.onCreateView(LayoutInflater.java:720)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:788)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
       at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
       at androidx.media2.widget.MusicView.inflateLayout(MusicView.java:144)
       at androidx.media2.widget.MusicView.<init>(MusicView.java:38)
       at androidx.media2.widget.VideoView.initialize(VideoView.java:322)
       at androidx.media2.widget.VideoView.<init>(VideoView.java:268)
       at androidx.media2.widget.VideoView.<init>(VideoView.java:263)
       at androidx.media2.widget.VideoView.<init>(VideoView.java:259)
       at com.mopub.mobileads.factories.VideoViewFactory.internalCreate(VideoViewFactory.java:18)
       at com.mopub.mobileads.factories.VideoViewFactory$Companion.create(VideoViewFactory.java:15)
       at com.mopub.mobileads.VastVideoViewController.createVideoView(VastVideoViewController.java:339)
       at com.mopub.mobileads.VastVideoViewController.<init>(VastVideoViewController.java:205)
       at com.mopub.mobileads.FullscreenAdController.createVideoViewController(FullscreenAdController.java:332)
       at com.mopub.mobileads.FullscreenAdController.<init>(FullscreenAdController.java:217)
       at com.mopub.mobileads.MoPubFullscreenActivity.onCreate(MoPubFullscreenActivity.java:72)
       at android.app.Activity.performCreate(Activity.java:7009)
       at android.app.Activity.performCreate(Activity.java:7000)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
       at android.app.ActivityThread.-wrap11(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6494)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by chauduyphanvu** [comment]
@m-melis, from the stack trace, it seems like the crash is happening when we're about to present a fullscreen ad experience and the `androidx.media2.widget.VideoView` can't look up some internal resources (an ImageView in this case). 

Do you have all of these [dependencies](https://github.com/mopub/mopub-android-sdk#requirements) added to your app? 

Vu Chau
MoPub Team

**Comment by m-melis** [comment]
Thanks for the answer. As I've integrated the SDK via jCenter AAR I've not explicitly added all the dependencies (https://developers.mopub.com/publishers/android/integrate/#option-1-jcenter-aar-recommended): "Add the dependencies listed here if you integrate the MoPub Android SDK as a standalone AAR, and not via Bintray/JCenter."

Is that correct? I'll try to add the explicit dependencies for `androidx.media2` anyway and see if the crash happens again.

**Comment by chauduyphanvu** [comment]
If you integrate via JCenter, then you do not need to add the other dependencies manually.

Back to the original issue, do you know if it also happens on other device/OS combinations besides the Nexus 6P + Android 8.1? Could you share the ad unit ID that serves the ad at that time?

**Comment by m-melis** [comment]
At this time we only found that occurrence of the crash (on 6 apr 2021, 13:56:54). Ad unit `3738aee86ab345c9a42b8f1c7764ef8e`

**Comment by chauduyphanvu** [comment]
@m-melis, thanks for the confirmation. Are you still getting this crash? 

I haven't been able to reproduce this crash using our sample app and available test devices, so the crash appears to be specific to that particular device and integration combination in your report. 

Feel free to share any additional details you might have, and we can take another look.

**Comment by chauduyphanvu** [comment]
Closing as there's been no new progress. Feel free to re-open if needed!

**Comment by ivan-goncharov-easybrain** [comment]
@chauduyphanvu, Hello!
We see the same problem in firebase crashlytcs and it happens on multiple different devices. Can you look into it once more? 

The stacktrace is similar, but in some reports we see the problematic layout resource which is `layout/music_with_title_landscape.xml`
```
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{xxx.yyy.zzz/com.mopub.mobileads.MoPubFullscreenActivity}: android.view.InflateException: Binary XML file line #24 in xxx.yyy.zzz:layout/music_with_title_landscape: Binary XML file line #24 in xxx.yyy.zzz/music_with_title_landscape: Error inflating class android.widget.ImageView
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3897)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4076)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:91)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2473)
       at android.os.Handler.dispatchMessage(Handler.java:110)
       at android.os.Looper.loop(Looper.java:219)
       at android.app.ActivityThread.main(ActivityThread.java:8349)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
```
```
Caused by android.content.res.Resources$NotFoundException: Unable to find resource ID #0x7f08016b
       at android.content.res.ResourcesImpl.getResourcePackageName(ResourcesImpl.java:387)
       at android.content.res.ResourcesImplEx.getResourcePackageName(ResourcesImplEx.java:50)
       at android.content.res.HwResourcesImpl.getThemePackageName(HwResourcesImpl.java:1742)
       at android.content.res.HwResourcesImpl.loadDrawable(HwResourcesImpl.java:1340)
       at android.content.res.ResourcesImpl.loadDrawable(ResourcesImpl.java:885)
       at android.content.res.Resources.loadDrawable(Resources.java:1004)
       at android.content.res.TypedArray.getDrawableForDensity(TypedArray.java:1207)
       at android.content.res.TypedArray.getDrawable(TypedArray.java:1182)
       at android.widget.ImageView.<init>(ImageView.java:213)
       at android.widget.ImageView.<init>(ImageView.java:194)
       at android.widget.ImageView.<init>(ImageView.java:190)
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at android.view.LayoutInflater.createView(LayoutInflater.java:864)
       at android.view.LayoutInflater.createView(LayoutInflater.java:786)
       at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:58)
       at com.android.internal.policy.HwPhoneLayoutInflater.onCreateView(HwPhoneLayoutInflater.java:102)
       at android.view.LayoutInflater.onCreateView(LayoutInflater.java:940)
       at android.view.LayoutInflater.onCreateView(LayoutInflater.java:960)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1014)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:971)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:1133)
       at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1094)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:692)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:536)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:483)
       at androidx.media2.widget.MusicView.inflateLayout(MusicView.java:144)
       at androidx.media2.widget.MusicView.<init>(MusicView.java:38)
       at androidx.media2.widget.VideoView.initialize(VideoView.java:279)
       at androidx.media2.widget.VideoView.<init>(VideoView.java:225)
       at androidx.media2.widget.VideoView.<init>(VideoView.java:220)
       at androidx.media2.widget.VideoView.<init>(VideoView.java:216)
       at com.mopub.mobileads.factories.VideoViewFactory.internalCreate(VideoViewFactory.java:18)
       at com.mopub.mobileads.factories.VideoViewFactory$Companion.create(VideoViewFactory.java:15)
       at com.mopub.mobileads.VastVideoViewController.createVideoView(VastVideoViewController.java:369)
       at com.mopub.mobileads.VastVideoViewController.<init>(VastVideoViewController.java:196)
       at com.mopub.mobileads.FullscreenAdController.createVideoViewController(FullscreenAdController.java:221)
       at com.mopub.mobileads.FullscreenAdController.<init>(FullscreenAdController.java:176)
       at com.mopub.mobileads.MoPubFullscreenActivity.onCreate(MoPubFullscreenActivity.java:74)
       at android.app.Activity.performCreate(Activity.java:8085)
       at android.app.Activity.performCreate(Activity.java:8073)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1320)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3870)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4076)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:91)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2473)
       at android.os.Handler.dispatchMessage(Handler.java:110)
       at android.os.Looper.loop(Looper.java:219)
       at android.app.ActivityThread.main(ActivityThread.java:8349)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
```

We also looked up the ID of allegedly missing resource and found it in 2 separate build versions -- different IDs from report correspond to the same drawable resource.

![image (3)](https://user-images.githubusercontent.com/56021456/131801251-9350944b-5bd4-43d4-becd-4066805b7b39.png)

**Comment by p158276** [comment]
Hi @ivan-goncharov-easybrain ,
Thanks for reaching out. Are there any specific android version that you saw this error? Besides, are you able to reproduce this issue? May I know if you declated any **androidx.media2.widget** dependencies in your Gradle file? 

Yoyo Lin 
MoPub Team

**Comment by ivan-goncharov-easybrain** [comment]
Hi @p158276 ,
Nothing really stands out. Vendors distribution looks like this:
- 38% HUAWEI
- 27% samsung
- 19% OPPO
- 12% motorola

and android versions distribution is the following:
- 65% Android 10
- 19% Android 8
- 12% Android 7
- 4% Android 11

We weren't seen this crash during our testing.

We don't have **media2** dependency in our scripts. I can even say that the only source of this library in our app is MoPub SDK - I had ran gradle's `dependencies` script to see if any other SDK depends on this library and found nothing. 

```
|    |    +--- com.mopub:mopub-sdk-base:5.17.0.4
|    |    |    +--- androidx.legacy:legacy-support-v4:1.0.0 (*)
|    |    |    +--- androidx.annotation:annotation:1.1.0 -> 1.2.0
|    |    |    +--- androidx.media2:media2-session:1.1.3 (*)
|    |    |    +--- androidx.media2:media2-widget:1.1.3 (*)
```
 resolved version of the library stays at 1.1.3

**Comment by chauduyphanvu** [comment]
@ivan-goncharov-easybrain, thanks for the additional details. 

It looks like this is coming from Google's media2 that we use to show fullscreen ads. Since we don't interact with media2 directly from our code, we recommend that you escalate this issue to Google for further assistance. It appears they are having issues inflating ImageView(s).
